core/src/main/java/jenkins/install/InstallState.java

```diff
@@ -196,8 +198,42 @@ public void initializeState() {
      * Upgrade of an existing Jenkins install.
      */
     @Extension
-    public static final InstallState UPGRADE = new UpgradeWizard();
-    
+    public static final InstallState UPGRADE = new Upgrade();
+    private static final class Upgrade extends InstallState {
+
+        Upgrade() {
+            super("UPGRADE", true);
+        }
+
+        @Override
+        public void initializeState() {
+            applyForcedChanges();
+
+            // Schedule an update of the update center after a Jenkins upgrade
+            reloadUpdateSiteData();
+        }
+
+        /**
+         * Put here the different changes that are enforced after an update.
+         */
+        private void applyForcedChanges(){
+            // Disable the legacy system of API Token only if the new system was not installed
+            // in such case it means there was already an upgrade before
+            // and potentially the admin has re-enabled the features
+            ApiTokenPropertyConfiguration apiTokenPropertyConfiguration = ApiTokenPropertyConfiguration.get();
+            if(!apiTokenPropertyConfiguration.hasExistingConfigFile()){
+                LOGGER.log(Level.INFO, "New API token system configured with insecure options to keep legacy behavior");
+                apiTokenPropertyConfiguration.setCreationOfLegacyTokenEnabled(false);
+                apiTokenPropertyConfiguration.setTokenGenerationOnCreationEnabled(false);
+            }
+        }
+
+    }
+
+    private static void reloadUpdateSiteData() {
+        Timer.get().submit(UpdateCenter::updateDefaultSite);
```

Comment:
why only update the default site?
also as the default site may not be available - this will cause a [`SEVERE`](https://github.com/jenkinsci/jenkins/blob/23bfee2fa93279e061553a1394dd6e5ddb967695/core/src/main/java/hudson/model/UpdateCenter.java#L2591-L2592) level log on every new version start (which is not good).

Reply:
> may not exist

Is there a reasonable configuration where that's the case? Offline instances, but they probably don't get new cores every week.

Reply:
@jtnord the setup wizard also calls this method:
https://github.com/jenkinsci/jenkins/blob/23bfee2fa93279e061553a1394dd6e5ddb967695/core/src/main/java/jenkins/install/SetupWizard.java#L214-L219

Can do other sites if required, I suspect the `SetupWizard` code should also use that as well?

Basically the reason I did it this way as the simplest API is `UpdateCenter::updateDefaultSite` which was probably written for `SetupWizard`

Reply:
CloudBees' products do not have an update site with the id `default`

When run at clean install time - the code to replace the `UpdateSite`s may not yet have run and so you may (or may not) get the log -> https://github.com/jenkinsci/jenkins/blob/23bfee2fa93279e061553a1394dd6e5ddb967695/core/src/main/java/hudson/model/UpdateCenter.java#L2590-L2593

thus for a clean install you may well get lucky more often than not.  
For an upgrade scenario this may be less likely and you may well be more likely to hit this. (currently the code re-adds the `default` site on every call to `load` but this is a bug I will probably address in the short term future.

But - even for an OSS user there is nothing special about the `default` update site.  The plugin you want to install may come from **any** Update Site - so treating just one of them as special is wrong,  either all should be updated or non of them.

As for only doing the default on a clean install - that was originally done (I guess) so that recommended plugins could be installed.  (and they for OSS only ever came through the default site.  (but updating all of them still would make some sense to me)

Reply:
sure will fix in a min